src/Features/CSharp/Portable/Formatting/CSharpFormattingInteractionService.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
```

Comment:
please tell me this was pure extraction and no change in logic (except where absolutely necessary for signatures).  If you did cahnge more than that, please call out where.  thanks :)

Reply:
If you go commit-by-commit you can see the individual changes. The only real changes were changing where the options come from and adding the inferred indentation service. Unfortunately, git didn't seem to follow the rename correctly, instead preferring to think that I renamed the CSharpEditorService back to the EditorFeatures layer and creating this file from whole cloth. Before merging I'll try to clean that up.

Reply:
To make reviewing this easier, I've attached the patch output of `git diff upstream/main:src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs src/Features/CSharp/Portable/Formatting/CSharpFormattingInteractionService.cs > actual.diff`, which shows the actual changes in the service.
[actualdiff.txt](https://github.com/dotnet/roslyn/files/6332358/actualdiff.txt)